kevm_pyk/src/kevm_pyk/__main__.py

```diff
@@ -20,149 +20,203 @@
+    for json_file in sorted(glob.glob(path_glob)):
+        _LOGGER.info(f'Processing contract file: {json_file}')
+        contract_name = json_file.split('/')[-1]
+        contract_name = contract_name[0:-5] if contract_name.endswith('.json') else contract_name
+        with open(json_file, 'r') as cjson:
+            contract_json = json.loads(cjson.read())
+            contract = Contract(contract_name, contract_json, foundry=True)
+            module, claims_module = contract_to_k(contract, empty_config, foundry=True, exclude_tests=_exclude_tests)
+            _LOGGER.info(f'Produced contract module: {module.name}')
+            modules.append(module)
+            if claims_module:
+                claims_modules.append(claims_module)
+    _main_module = KFlatModule(main_module, [], [KImport(mname) for mname in [_m.name for _m in modules] + imports])
+    _spec_module = KFlatModule(spec_module, [], [KImport(mname) for mname in [_m.name for _m in claims_modules]])
+    modules.append(_main_module)
+    modules.append(_spec_module)
+    bin_runtime_definition = KDefinition(_main_module.name, modules + claims_modules, requires=[KRequire(req) for req in ['edsl.md', 'lemmas/int-simplification.k', 'lemmas/lemmas.k'] + requires])
+    _kprint = KPrint_make_unparsing(kevm, extra_modules=modules)
+    KEVM._patch_symbol_table(_kprint.symbol_table)
+    print(_kprint.pretty_print(bin_runtime_definition) + '\n')
+
+
+def exec_prove(
+    definition_dir: Path,
+    profile: bool,
+    spec_file: Path,
+    includes: List[str],
+    debug_equations: List[str],
+    bug_report: bool,
+    spec_module: Optional[str],
+    **kwargs,
+) -> None:
+    kevm = KEVM(definition_dir, profile=profile)
+    prove_args = add_include_arg(includes)
+    haskell_args = []
+    for de in debug_equations:
+        haskell_args += ['--debug-equation', de]
+    if bug_report:
+        haskell_args += ['--bug-report', str(spec_file.with_suffix(''))]
+    final_state = kevm.prove(spec_file, spec_module_name=spec_module, args=prove_args, haskell_args=haskell_args)
+    print(kevm.pretty_print(final_state) + '\n')
+
+
+def exec_run(
+    definition_dir: Path,
+    profile: bool,
+    input_file: Path,
+    term: bool,
+    parser: Optional[str],
+    expand_macros: str,
+    depth: Optional[int],
+    output: str,
+    **kwargs,
+) -> None:
+    kevm = KEVM(definition_dir, profile=profile)
+    krun_args = []
+    if term:
+        krun_args += ['--term']
+    if parser is not None:
+        krun_args += ['--parser', parser]
+    if not expand_macros:
+        krun_args += ['--no-expand-macros']
+    # TODO: These are inlined into _krun
+    krun_args += ['--output', output]
+    krun_result = _krun(kevm.definition_dir, Path(input_file), depth=depth, args=krun_args, profile=profile)
+    print(krun_result.stdout)
+    sys.exit(krun_result.returncode)
+
+
+# Helpers
+
+def _create_argument_parser() -> ArgumentParser:
 
```

Comment:
Is there a reason for the blank lines?

Reply:
Between functions, it is to conform to PEP8 code style: https://peps.python.org/pep-0008/#blank-lines (this is objective).

Within functions, it is to make it easier to parse visually and separate logical units (this is subjective).